app/src/main/java/com/courseproject/tindar/ui/chat/ChatActivity.java

```diff
@@ -86,13 +88,14 @@ protected void onCreate(Bundle savedInstanceState) {
         // This "came with the class." I assume it's right and it works, so I'm not touching it.
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_chat); // tells us the layout follows activity_chat.xml
+        Intent intent = this.getIntent();
 
         // TODO: actually get info passed in, rather than hardcoding this
         // TODO: see if we can have this info passed and stored in some sort of facade or model
         //      - flagged in issue #51
-        this.userID = "user1";
-        this.otherUserID = "user2";
-        this.conversationPartnerDisplayName = "User 2";
+        this.userID = intent.getStringExtra("current_user_id");
+        this.otherUserID = intent.getStringExtra("conversation_partner_id");
+        this.conversationPartnerDisplayName = this.otherUserID;
```

Comment:
Can we use useId / otherUserId instead of userID / otherUserID to make the naming convention consistent?

why displayName is otherUserId but not otherUserDisplayName? That information is in the matchListFragment so you can pass it from there.

Reply:
First one is done. Second one: I'll grab a profile model from the database when I actually implement this. Grabbing it from the matchlist fragment is not actually as straightforward as one might expect.